### PR TITLE
Add AI image block

### PR DESCRIPTION
**Pull request by CiaranMn**
## 🌟 What is the purpose of this PR?

Introduces a new block which uses the Service Module to request AI image generation from OpenAI.

## 🔗 Related links
[Asana task](https://app.asana.com/0/1203358502199087/1203553091426341/f) (internal)

## 🔍 What does this change?
- Adds the block

## 🐾 Next steps
- Publish
- Allow users to select the number of images generated (internal [task](https://app.asana.com/0/0/1203845052955898/f))

## ❓How to test this?
1. yarn && cd blocks/ai-text && yarn dev
2. add a valid Block Protocol API key to src/dev.tsx
3. Try generating some text

### CODE REVIEW

**Review by nonparibus — APPROVED** [review]
Legal ✅

File: blocks/ai-image/src/app/generate-image.tsx [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+    ImageObject[] | RemoteFileEntity[] | null
+  >(null);
+
+  const uploadInProgress = images?.some((image) => !isFileEntity(image));
+
+  const confirm = (imageEntityId: string) => {
+    void graphModule.createEntity({
+      data: {
+        entityTypeId:
+          "https://blockprotocol-87igvkbkw.stage.hash.ai/@ciaranm/types/entity-type/generatedimage/v/1",
```
Comment (luisbettencourt) [inline blocks/ai-image/src/app/generate-image.tsx:44]: Does this entityTypeId need updating?

**Review by luisbettencourt — CHANGES_REQUESTED** [review]

File: blocks/ai-image/src/dev.tsx [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+ * 2. It holds an in-memory datastore of entities and links
+ * 3. It listens for messages from your blocks and updates its datastore appropriately (e.g. to create a new entity)
+ * 4. It displays a debug UI allowing you to see the contents of its datastore, and messages sent back and forth
+ */
+const DevApp = () => {
+  return (
+    <MockBlockDock
+      blockDefinition={{ ReactComponent: Component }}
+      blockInfo={packageJSON.blockprotocol}
+      blockProtocolApiKey={undefined} // Set this to an API key when testing
```
[the remark this answers is not in the document]
Comment (CiaranMn) [reply blocks/ai-image/src/dev.tsx:26]: Will address in follow up

File: blocks/ai-image/src/app/generate-image.tsx (hunk 1 above)
Comment (CiaranMn) [reply blocks/ai-image/src/app/generate-image.tsx]: 🙈 thank you!

**Review by luisbettencourt — APPROVED** [review]

**Pull request merged**